contracts/src/lib/LibStaking.sol

```diff
@@ -611,15 +586,15 @@ library LibValidatorTracking {
         ParentValidatorsTracker storage self,
         uint16 activeValidatorsLimit,
         GenesisValidator[] memory validators
-    ) internal returns (Validator[] memory membership) {
+    ) internal returns (Validator[] memory) {
         self.validators.activeLimit = activeValidatorsLimit;
         // Start the next configuration number from 1, 0 is reserved for no change and the genesis membership
         self.changes.nextConfigurationNumber = LibStaking.INITIAL_CONFIGURATION_NUMBER;
         // The startConfiguration number is also 1 to match with nextConfigurationNumber, indicating we have
         // empty validator change logs
         self.changes.startConfigurationNumber = LibStaking.INITIAL_CONFIGURATION_NUMBER;
 
-        initValidators(self, validators);
+        return initValidators(self, validators);
```

Comment:
Did Solidity not realise that there was nothing returned before :scream: ?

Reply:
yeah, that's the thing, solidity does not report an error, it's very easy to make a mistake, lol.

Reply:
What's worse is that none of the 100 linters, static analyzers, nor auditors caught this 💩